etna/models/nn/deepstate/deepstate.py

```diff
@@ -180,8 +212,16 @@ def forward(self, x: DeepStateBatch, *args, **kwargs):  # type: ignore
 
     def make_samples(self, df: pd.DataFrame, encoder_length: int, decoder_length: int) -> Iterator[dict]:
         """Make samples from segment DataFrame."""
-        values_real = df.drop(columns=["target", "segment", "timestamp"]).values
+        values_real = df.drop(columns=["target", "segment", "timestamp"] + list(self.embedding_sizes.keys())).values
         values_real = torch.from_numpy(values_real).float()
+
+        # Categories that were not seen during `fit` will be filled with new category
+        for feature in self.embedding_sizes:
```

Comment:
Don't we want may be add this logic into some other class for example?

Reply:
You mean for not nn models?

Reply:
I mean we could, for example, move this logic into our embedding class or smth like this.
We could try to do this later, not in this PR.